modules/protocol/src/errors.ts

```diff
@@ -66,6 +66,7 @@ export class CreateUpdateError extends ProtocolError {
     CouldNotSign: "Failed to sign updated channel hash",
     FailedToReconcileDeposit: "Could not reconcile deposit",
     FailedToResolveTransferOnchain: "Could not resolve transfer onchain",
+    FailedToUpdateMerkleRoot: "Could not generate new merkle root",
```

Comment:
This seems really unlikely. The only reason this should ever throw is if there is a bug.

Reply:
Yeah the issue is more that *if* it fails due to a bug it shouldn't throw an error it should return a `Result.fail(...)` so it can be handled properly at the higher levels